src/main/java/util/Constants.java

```diff
@@ -108,6 +108,8 @@
     public static final int USER_LOCK_TIMEOUT = 21;
     // 15 minutes in milliseconds
     public static final int LOCK_DURATION = 60 * 15 * 1000;
+    public static final int CONNECT_TIMEOUT = 60 * 1000;
```

Comment:
The documentation implies that this connection timeout is for "establishing a connection," which sounds correct to me although I'm not confident in when Java has decided that has occurred . Did you test this against an HQ restore which took more than 60 seconds to return to ensure that it doesn't timeout if it hasn't received a response code within this duration?

Reply:
I did not test it against an actual restore. My understanding of [the documentation](https://docs.oracle.com/javase/8/docs/api/java/net/URLConnection.html#setConnectTimeout-int-)

> If the timeout expires before the connection can be established, a java.net.SocketTimeoutException is raised.

is that it is the amount of time allowed for the connection to reach the ESTABLISHED state, which happens to be the state of the hung connections. This [SO answer](https://stackoverflow.com/a/3069450/10840) seems pretty clear.

It sounds like you're wondering if `connectTimeout` might be the total amount of time to complete the request, but nothing I've read has implied anything like that. Are you concerned that the implementation differs from the documentation?